emcc.py

```diff
@@ -580,7 +580,14 @@ def uniquename(name):
 
   temp_root = shared.TEMP_DIR
   if not os.path.exists(temp_root):
-    os.makedirs(temp_root)
+    try:
+      os.makedirs(temp_root)
+    except Exception as e:
+      if hasattr(e, 'winerror') and e.winerror == 183 and os.path.exists(temp_root): # https://msdn.microsoft.com/en-us/library/windows/desktop/ms681382(v=vs.85).aspx: 183 == ERROR_ALREADY_EXISTS
```

Comment:
do we need the error detection? it seems like if we hit an error, and then see the dir exists, we can continue?

Reply:
Thought about the same, but felt it might be a bit brutish to possibly mute any errors. However re-thinking today, I find it has the appeal of working on all OSes without tricks, so updated the PR to go that way. How does this look now?